### PR TITLE
Doc Improvement - RMSProp and RMSPropAlex

### PR DESCRIPTION
**Pull request by jiajiechen**
Add description for RMSProp update and RMSPropAlex update in `optimizer_op.cc`

@mli @zackchase @madjam @nswamy

### CODE REVIEW

File: python/mxnet/optimizer.py [hunk 1]
```diff
@@ -541,25 +546,27 @@ class RMSProp(Optimizer):
...
     ----------
     gamma1: float, optional
-        Decay factor of moving average for ``gradient^2``.
+        A decay factor of moving average over past squared gradient.
     gamma2: float, optional
-        A "momentum" factor. Only used if ``centered=True``.
+        A "momentum" factor. Only used if `centered`=``True``.
     epsilon : float, optional
-        Small value to avoid division by 0.
+        Small value to avoid divided by 0.
```
Comment (madjam) [inline python/mxnet/optimizer.py]: division is the right word here.

File: python/mxnet/optimizer.py [hunk 2]
```diff
@@ -541,25 +546,27 @@ class RMSProp(Optimizer):
...
     centered : bool, optional
-        Use Graves' or Tieleman & Hinton's version of RMSProp.
+        Uses Graves's version of `RMSProp` or Tieleman & Hinton's version of `RMSProp`.
```
Comment (madjam) [inline python/mxnet/optimizer.py]: you should reword this to say:
Flag to control which version of RMSProp to use. ``True`` will use X, ``False`` will use Y

File: src/operator/optimizer_op.cc [hunk 3]
```diff
@@ -105,9 +106,35 @@ It updates the weights using::
...
 .add_arguments(AdamParam::__FIELDS__());
 
+
 NNVM_REGISTER_OP(rmsprop_update)
-.describe(R"code(Update function for RMSProp optimizer. The RMSProp code follows the version in
-http://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf Tieleman & Hinton, 2012.
+.describe(R"code(Update function for `RMSProp` optimizer.
+
+The `RMSprop` is a form of stochastic gradient descent where the gradients are
+divided by a running average of their recent magnitude. `RMSprop` was developped
```
Comment (madjam) [inline src/operator/optimizer_op.cc]: developed typo

File: src/operator/optimizer_op.cc [hunk 4]
```diff
@@ -105,9 +106,35 @@ It updates the weights using::
...
+from the need to resolve `Adagrad`'s radically diminishing learning rates.
```
Comment (madjam) [inline src/operator/optimizer_op.cc]: omit "from the need"

File: src/operator/optimizer_op.cc [hunk 5]
```diff
@@ -125,8 +152,28 @@ NNVM_REGISTER_OP(rmsprop_update)
 .add_arguments(RMSPropParam::__FIELDS__());
 
 NNVM_REGISTER_OP(rmspropalex_update)
-.describe(R"code(Update function for RMSPropAlex optimizer. The RMSPropAlex code follows the version in
+.describe(R"code(Update function for RMSPropAlex optimizer.
+
+The `RMSPropAlex` is another version of `RMSProp`.
```
Comment (piiswrong) [inline src/operator/optimizer_op.cc]: non-centered version?

File: src/operator/optimizer_op.cc [hunk 6]
```diff
@@ -105,9 +106,35 @@ It updates the weights using::
 .add_argument("var", "NDArray-or-Symbol", "Moving variance")
 .add_arguments(AdamParam::__FIELDS__());
 
+
 NNVM_REGISTER_OP(rmsprop_update)
-.describe(R"code(Update function for RMSProp optimizer. The RMSProp code follows the version in
-http://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf Tieleman & Hinton, 2012.
+.describe(R"code(Update function for `RMSProp` optimizer.
+
+The `RMSprop` is a form of stochastic gradient descent where the gradients are
```
Comment (zackchase) [inline src/operator/optimizer_op.cc]: "The `RMSProp` is ..." sounds strange to me. Please change to "`RMSProp` is a variant of..."

File: src/operator/optimizer_op.cc (hunk 3 above)
Comment (zackchase) [inline src/operator/optimizer_op.cc]: This is factually incorrect - right? Aren't they divided by a cache which grows with the sum of squares of recent gradients?

File: src/operator/optimizer_op.cc (hunk 4 above)
Comment (zackchase) [inline src/operator/optimizer_op.cc]: This could be clearer. 
Instead of "`RMSprop` was developed from the need to resolve `Adagrad`'s radically diminishing learning rates."

Please write:

"`RMSProp` is similar to AdaGrad, a popular variant of SGD which adaptively tunes the learning rate of each parameter. AdaGrad lowers the learning rate for each parameter monotonically over the course of training. While this is analytically motivated for convex optimizations, it may not be ideal for non-convex problems. RMSProp deals with this heuristically by allowing the learning rates to rebound as the denominator decays over time."

File: src/operator/optimizer_op.cc [hunk 7]
```diff
@@ -125,8 +152,28 @@ NNVM_REGISTER_OP(rmsprop_update)
...
+
+The update step is
+
+.. math::
+  \theta_{t+1} = \theta_t + \Delta_t
+
+The RMSPropAlex code follows the version in
 http://arxiv.org/pdf/1308.0850v5.pdf Eq(38) - Eq(45) by Alex Graves, 2013.
+
+Alex suggests the momentum term :math:`\gamma_1` to be 0.95, :math:`\gamma_2`
```
Comment (zackchase) [inline src/operator/optimizer_op.cc]: Let's not refer to researchers by their first names. Change to "Graves suggests that the momentum term..."

File: src/operator/optimizer_op.cc (hunk 5 above)
Comment (jiajiechen) [inline src/operator/optimizer_op.cc]: thanks for suggestion!

File: src/operator/optimizer_op.cc (hunk 6 above)
Comment (jiajiechen) [inline src/operator/optimizer_op.cc]: Thanks!

File: src/operator/optimizer_op.cc (hunk 3 above)
Comment (jiajiechen) [inline src/operator/optimizer_op.cc]: @zackchase isnt the cache growing with square root of the sum of squares of recent gradients?

File: src/operator/optimizer_op.cc (hunk 4 above)
Comment (jiajiechen) [inline src/operator/optimizer_op.cc]: Thanks!
